nb2/bot/slack_bot.py

```diff
@@ -194,113 +241,177 @@ def remember(self, target_slack_user_id: str, quote: str):
 
         return self.Result(ok=True, message="Memory stored!")
 
-    def _random_quote(self, nostalgia_user_target):
+    def _random_quote(self, nostalgia_user_target: str = None) -> Tuple[Union[Person, str], Quote]:
         """
         Fetch a random quote from NB's memory of a Person with nostalgia_user_target, if they exist.
+        If no user is specified, pick a random Quote and return that user and quote.
 
         Args:
-            nostalgia_user_target: string representing slack user id for Person
-                                   to retrieve quote.
+            nostalgia_user_target: string representing slack_user_id, ghost_user_id or display_name
+                                   for Person to retrieve quote.
+                                   If not provided, this will return a random quote from
+                                   a random Person.
 
         Returns:
-            A tuple with a person's name and a Quote (or None if no quotes were found)
+            A tuple. If the Person doesn't exist, the tuple will be a string with a string
```

Comment:
I know this note explains why the first element is sometimes a string (nostalgia_user_target) and not a Person but it's confusing for me when we actually use the person and quote returned by this function.

Could we maybe change this to always return either nostalgia_user_target or person.first_name instead of a person. It seems like we either use the person returned as if they are a user_id or a person object that has a first name

Reply:
Looks like that's the case yup, I think we can just keep the first item always as a str (target / first name)

Reply:
Updated, I also changed the second item in the tuple to return the quote's content for similar reasons - we always just use the quote content in our code, so this tuple would either be (str, str) or (str, None)

Reply:
Makes perfect sense, and I checked the changes :+1: